libs/src/api/serviceProvider.ts

```diff
@@ -122,34 +130,53 @@ class ServiceProvider extends Base {
    * @param {any[]} discoveryNodes the verbose list of discovery nodes to select from
    * @param {(node: { delegateOwnerWallet: string }) => boolean} filter an optional filter step to remove certain nodes
    */
-  async getUniquelyOwnedDiscoveryNodes ({ quorumSize, discoveryNodes = [], filter = (node) => true, useWhitelist = true }) {
+  async getUniquelyOwnedDiscoveryNodes({
+    quorumSize,
+    discoveryNodes = [],
+    filter = (_) => true,
+    useWhitelist = true
+  }: {
+    quorumSize: number
+    discoveryNodes?: ServiceWithEndpoint[]
+    filter: (node: ServiceWithEndpoint) => boolean
+    useWhitelist?: boolean
+  }) {
     if (!discoveryNodes || discoveryNodes.length === 0) {
       // Whitelist logic: if useWhitelist is false, pass in null to override internal whitelist logic; if true, pass in undefined
       // so service selector uses internal whitelist
-      discoveryNodes = await this.discoveryProvider.serviceSelector.findAll({ verbose: true, whitelist: useWhitelist ? undefined : null })
+      discoveryNodes = (await this.discoveryProvider.serviceSelector.findAll({
+        verbose: true,
+        whitelist: useWhitelist ? undefined : null
+      })) as ServiceWithEndpoint[]
     }
 
     discoveryNodes.filter(filter)
 
     // Group nodes by owner
-    const grouped = discoveryNodes.reduce((acc, curr) => {
+    const grouped = discoveryNodes.reduce<{
+      [owner: string]: ServiceWithEndpoint[]
+    }>((acc, curr) => {
       if (curr.owner in acc) {
+        // @ts-expect-error
```

Comment:
Couldn't get this to accept that acc[cur.owner] was not undefined within the branch...

Reply:
yeah i guess cause it assumes there is a chance that the object can be modified on another thread or something crazy? well i think two options are:
```
      const currentEndpoints = acc[curr.owner]
      if (currentEndpoints) {
        currentEndpoints.push(curr)
      } else {
        acc[curr.owner] = [curr]
      }
```
or 
```
      if (curr.owner in acc) {
        acc[curr.owner]?.push(curr)
      } else {
        acc[curr.owner] = [curr]
      }
      return acc
```
and this one is a bit different, but see what you think:
```
      acc[curr.owner] = [...(acc[curr.owner] ?? []), curr]
      return acc

Reply:
great options! went with #2 for clarity :)